lessons/cyclesort.md

```diff
@@ -0,0 +1,77 @@
+---
+path: "Cyclic"
+title: "Cyclic Sort"
+order: "5E"
+section: "Searching & Sorting"
+description: "learn Sorting algorithms"
+---
+
+
+Cycle sort is a comparison sorting algorithm which forces array to be factored into the number of cycles where each of them can be rotated to produce a sorted array. It is theoretically optimal in the sense that it reduces the number of writes to the original array.
+
+
+It is an in-place and unstable sorting algorithm.
+
+It is optimal in terms of number of memory writes. It minimizes the number of memory writes to sort. Each value is either written zero times, if it’s already in its correct position, or written one time to its correct position.
+
+
+## Algorithm
+Consider an array of n distinct elements. An element a is given, index of a can be calculated by counting the number of elements that are smaller than a.
+
+ 1. if the element is found to be at its correct position, simply leave it as it is.
+ 2. Otherwise, find the correct position of a by counting the total number of elements that are less than a. where it must be present in the sorted array. The other element b which is replaced is to be moved to its correct position. This process continues until we got an element at the original position of a.
```

Comment:
```suggestion
 2. Otherwise, find the correct position of `a` by counting the total number of elements that are less than `a`. where it must be present in the sorted array. The other element `b` which is replaced is to be moved to its correct position. This process continues until we got an element at the original position of `a`.
```